siphon/acis.py

```diff
@@ -0,0 +1,51 @@
+import json
+import urllib.request
+import urllib.parse
+import json
+import socket
+
+def acisRequest(method, params):
+    """
+    
+    This function will make a request to the ACIS Web Services API for data
+    based on the given method (StnMeta,StnData,MultiStnData,GridData,General)
+    and parameters string. Information about the parameters can be obtained at:
+    http://www.rcc-acis.org/docs_webservices.html
+
+    If a connection to the API fails, then it returns False. A failed request
+    will have an error text in the return data produced by the remote API.
+
+    Parameters
+    ----------
+    method - The Web Services request method (StnMeta, StnData, MultiStnData, etc)
+    params - A dictionary of parameters (see API documentation)
+
+    Returns
+    ----------
+    Connection Success: A dictionary that is derived from JSON data from the remote API
+    Connection Failure: False
+
+    """
+    params = json.dumps(params).encode('utf8')
+
+    base_url = 'http://data.rcc-acis.org/'  # ACIS Web API URL
+
+    req = urllib.request.Request(base_url+method,
```

Comment:
Do you think you could adapt this to use the `requests` module? We're already using it internally, it would solve your Python 2 problem, and it sets our own "User-Agent" header. I'm happy to look into how to do this if you can clarify how the `data` argument gets used.

Reply:
The data argument is simply a JSON object containing parameters (Station ID, dates, variables, etc) for the HTTP request. I'm more than happy to adapt it, there was no reason for using urllib other than that's what was used in the old Python 2 version of this code.

Reply:
Ok, then I'll leave you to it, but feel free to scream if you run into problems.